### PR TITLE
[Fix] Fix the pytorch2onnx same as mmdet.

### PR DESCRIPTION
**Pull request by SuX97**

### CODE REVIEW

**Comment by innerlee** [comment]
how to deal with the previous version?

**Comment by SuX97** [comment]
> how to deal with the previous version?

will remove it.

**Comment by innerlee** [comment]
do it in this pr is ok

File: tools/pytorch2onnx.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    return module_output
+
+
+def pytorch2onnx(model,
+                 input_shape,
+                 opset_version=11,
+                 show=False,
+                 output_file='tmp.onnx',
+                 verify=False,
+                 normalize_cfg=None):
```
Comment (dreamerlin) [inline tools/pytorch2onnx.py:45]: Need docstring.

File: tools/pytorch2onnx.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        # get pytorch output
+        pytorch_result = model([one_img])
+
+        # get onnx output
+        input_all = [node.name for node in onnx_model.graph.input]
+        input_initializer = [
+            node.name for node in onnx_model.graph.initializer
+        ]
+        net_feed_input = list(set(input_all) - set(input_initializer))
+        assert (len(net_feed_input) == 1)
```
Comment (dreamerlin) [inline tools/pytorch2onnx.py:84]: outer brackets is redundant

File: tools/pytorch2onnx.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+        # only compare a part of result
+        assert np.allclose(
+            pytorch_result[0][:, 4], onnx_result[:, 4]
+        ), 'The outputs are different between Pytorch and ONNX'
+        print('The numerical values are same between Pytorch and ONNX')
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description='Convert MMAction models to ONNX')
```
Comment (dreamerlin) [inline tools/pytorch2onnx.py:97]: MMAction2

**Comment by SuX97** [comment]
@dreamerlin @innerlee This PR is ready, checkpoints have been checked.

File: tools/pytorch2onnx.py [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+try:
+    import onnx
+    import onnxruntime as rt
+except ImportError as e:
+    raise ImportError(f'Please install onnx and onnxruntime first. {e}')
+
+try:
+    from mmcv.onnx.symbolic import register_extra_symbolics
+except ModuleNotFoundError:
+    raise NotImplementedError('please update mmcv to version>=v1.0.4')
```
Comment (dreamerlin) [inline tools/pytorch2onnx.py:19]: `version>=1.0.4`

File: tools/pytorch2onnx.py [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+
+def pytorch2onnx(model,
+                 input_shape,
+                 opset_version=11,
+                 show=False,
+                 output_file='tmp.onnx',
+                 verify=False):
+    """Convert pytorch model to onnx model.
+
+    Args:
```
Comment (dreamerlin) [inline tools/pytorch2onnx.py:52]: missing `model`

File: tools/pytorch2onnx.py [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+        default=[1, 3, 8, 224, 224],
+        help='input video size')
+    args = parser.parse_args()
+    return args
+
+
+if __name__ == '__main__':
+    args = parse_args()
+
+    assert args.opset_version == 11, 'MMAction2 only support opset 11 now'
```
Comment (dreamerlin) [inline tools/pytorch2onnx.py:132]: supports

**Comment by innerlee** [comment]
is this good to go?

can this be tested? making sure that the script is okay for a tiny tiny model is enough for the test

**Comment by SuX97** [comment]
Yes, I have tested almost all checkpoints, including TSN TIN SLOWFAST R2+1D BSN-tem. It's good to go.
________________________________

**Comment by SuX97** [comment]
> is this good to go?
> 
> can this be tested? making sure that the script is okay for a tiny tiny model is enough for the test

We need a checkpoint downloaded on the server(the smallest one we have is resnet-50 TSN), or `wget` when unittesting, onnx onnxruntime is needed for test, kinda heavy test to do.

**Comment by innerlee** [comment]
for test purpose, there is no need to use a real model.
a mini model with 1 conv 1 bn and 1 channel is good enough.
"the script can run" is what we want to test, not that "the result is correct".

File: docs/tutorials/export_model.md [hunk 7]
```diff
@@ -13,29 +13,38 @@ Open Neural Network Exchange [(ONNX)](https://onnx.ai/) is an open ecosystem tha
...
 ```
 
-### Recognizers
-For recognizers, if your model are trained with a config from mmaction2 and intend to inference it according to the test pipeline, simply run:
+We provide a python script to export the pytorch model trained by MMaction2 to ONNX.
 ```shell
-python tools/torch2onnx.py $CONFIG_PATH $CHECKPOINT_PATH
+python tools/pytorch2onnx.py ${CONFIG_FILE} ${CHECKPOINT_FILE} [--shape ${SHAPE}] \
+    [--verify] [--show] [--output-file ${OUTPUT_FILE}]  [--is-localizer] [--opset-version ${VERSION}]
+\
```
Comment (dreamerlin) [inline docs/tutorials/export_model.md:28]: redundant `\`

**Review by innerlee — APPROVED** [review]

**Pull request merged**